[Split Build] Test split build in pull CI workflow

PaliC opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #126813
* #127934

This PR builds the split build in the pull workflow and runs the appropriate tests against them. A single linux cpu and single gpu build were chosen arbitrarily to not add too many tests.

PaliC committed:
Update

[ghstack-poisoned]

PaliC committed:
Update

[ghstack-poisoned]

PaliC committed:
Update

[ghstack-poisoned]

PaliC committed:
Update

[ghstack-poisoned]

PaliC committed:
Update

[ghstack-poisoned]

PaliC committed:
Update

[ghstack-poisoned]

PaliC committed:
Update on "[Split Build] Test split build in pull CI workflow"


This PR builds the split build in the pull workflow and runs the appropriate tests against them. A single linux cpu and single gpu build were chosen arbitrarily to not add too many tests.

[ghstack-poisoned]

PaliC committed:
Update

[ghstack-poisoned]

PaliC committed:
Update

[ghstack-poisoned]

PaliC committed:
Update

[ghstack-poisoned]

PaliC committed:
Update

[ghstack-poisoned]

PaliC committed:
Update

[ghstack-poisoned]

PaliC committed:
Update

[ghstack-poisoned]

PaliC committed:
Update

[ghstack-poisoned]

PaliC commented on .github/actions/linux-build/action.yml:
```diff
@@ -192,14 +192,24 @@ runs:
         path: artifacts.zip
         s3-bucket: ${{ inputs.s3-bucket }}
 
+    - name: Store PyTorch Build Artifacts on S3 for split build
```
Otherwise we have a name collision with 2 jobs uploading to the S3 bucket. This means that one of the two jobs would have the wrong artifacts.

PaliC committed:
Update

[ghstack-poisoned]

PaliC committed:
Update

[ghstack-poisoned]

PaliC committed:
Update on "[Split Build] Test split build in pull CI workflow"


This PR builds the split build in the pull workflow and runs the appropriate tests against them. A single linux cpu and single gpu build were chosen arbitrarily to not add too many tests.

[ghstack-poisoned]

PaliC committed:
Update

[ghstack-poisoned]

PaliC committed:
Update

[ghstack-poisoned]

PaliC committed:
Update

[ghstack-poisoned]

PaliC committed:
Update

[ghstack-poisoned]

PaliC committed:
Update

[ghstack-poisoned]

PaliC committed:
Update

[ghstack-poisoned]

PaliC committed:
Update

[ghstack-poisoned]

PaliC committed:
Update

[ghstack-poisoned]

PaliC committed:
Update

[ghstack-poisoned]

PaliC committed:
Update

[ghstack-poisoned]

PaliC committed:
Update

[ghstack-poisoned]

PaliC committed:
Update

[ghstack-poisoned]

PaliC committed:
Update

[ghstack-poisoned]

PaliC committed:
Update

[ghstack-poisoned]

PaliC committed:
Update

[ghstack-poisoned]

PaliC committed:
Update

[ghstack-poisoned]

PaliC committed:
Update on "[Split Build] Test split build in pull CI workflow"


This PR builds the split build in the pull workflow and runs the appropriate tests against them. A single linux cpu and single gpu build were chosen arbitrarily to not add too many tests.

[ghstack-poisoned]

PaliC committed:
Update

[ghstack-poisoned]

PaliC committed:
Update

[ghstack-poisoned]

PaliC committed:
Update

[ghstack-poisoned]

PaliC committed:
Update

[ghstack-poisoned]

PaliC committed:
Update

[ghstack-poisoned]

PaliC committed:
Update

[ghstack-poisoned]

PaliC committed:
Update

[ghstack-poisoned]

PaliC committed:
Update

[ghstack-poisoned]

PaliC committed:
Update

[ghstack-poisoned]

PaliC committed:
Update

[ghstack-poisoned]

PaliC committed:
Update

[ghstack-poisoned]

PaliC committed:
Update

[ghstack-poisoned]

PaliC committed:
Update

[ghstack-poisoned]

PaliC committed:
Update on "[Split Build] Test split build in pull CI workflow"


This PR builds the split build in the pull workflow and runs the appropriate tests against them. A single linux cpu and single gpu build were chosen arbitrarily to not add too many tests.

[ghstack-poisoned]

PaliC committed:
Update on "[Split Build] Test split build in pull CI workflow"


This PR builds the split build in the pull workflow and runs the appropriate tests against them. A single linux cpu and single gpu build were chosen arbitrarily to not add too many tests.

[ghstack-poisoned]

PaliC committed:
Update on "[Split Build] Test split build in pull CI workflow"


This PR builds the split build in the pull workflow and runs the appropriate tests against them. A single linux cpu and single gpu build were chosen arbitrarily to not add too many tests.

[ghstack-poisoned]

PaliC committed:
Update on "[Split Build] Test split build in pull CI workflow"


This PR builds the split build in the pull workflow and runs the appropriate tests against them. A single linux cpu and single gpu build were chosen arbitrarily to not add too many tests.

[ghstack-poisoned]

PaliC committed:
Update on "[Split Build] Test split build in pull CI workflow"


This PR builds the split build in the pull workflow and runs the appropriate tests against them. A single linux cpu and single gpu build were chosen arbitrarily to not add too many tests.

[ghstack-poisoned]

PaliC committed:
Update on "[Split Build] Test split build in pull CI workflow"


This PR builds the split build in the pull workflow and runs the appropriate tests against them. A single linux cpu and single gpu build were chosen arbitrarily to not add too many tests.

[ghstack-poisoned]

PaliC committed:
Update on "[Split Build] Test split build in pull CI workflow"


This PR builds the split build in the pull workflow and runs the appropriate tests against them. A single linux cpu and single gpu build were chosen arbitrarily to not add too many tests.

[ghstack-poisoned]

PaliC committed:
Update on "[Split Build] Test split build in pull CI workflow"


This PR builds the split build in the pull workflow and runs the appropriate tests against them. A single linux cpu and single gpu build were chosen arbitrarily to not add too many tests.

[ghstack-poisoned]

PaliC committed:
Update on "[Split Build] Test split build in pull CI workflow"


This PR builds the split build in the pull workflow and runs the appropriate tests against them. A single linux cpu and single gpu build were chosen arbitrarily to not add too many tests.

[ghstack-poisoned]

PaliC committed:
Update on "[Split Build] Test split build in pull CI workflow"


This PR builds the split build in the pull workflow and runs the appropriate tests against them. A single linux cpu and single gpu build were chosen arbitrarily to not add too many tests.

[ghstack-poisoned]

PaliC committed:
Update on "[Split Build] Test split build in pull CI workflow"


This PR builds the split build in the pull workflow and runs the appropriate tests against them. A single linux cpu and single gpu build were chosen arbitrarily to not add too many tests.

[ghstack-poisoned]

PaliC committed:
Update on "[Split Build] Test split build in pull CI workflow"


This PR builds the split build in the pull workflow and runs the appropriate tests against them. A single linux cpu and single gpu build were chosen arbitrarily to not add too many tests.

[ghstack-poisoned]

PaliC committed:
Update on "[Split Build] Test split build in pull CI workflow"


This PR builds the split build in the pull workflow and runs the appropriate tests against them. A single linux cpu and single gpu build were chosen arbitrarily to not add too many tests.

[ghstack-poisoned]

PaliC committed:
Update on "[Split Build] Test split build in pull CI workflow"


This PR builds the split build in the pull workflow and runs the appropriate tests against them. A single linux cpu and single gpu build were chosen arbitrarily to not add too many tests.

[ghstack-poisoned]

PaliC committed:
Update on "[Split Build] Test split build in pull CI workflow"


This PR builds the split build in the pull workflow and runs the appropriate tests against them. A single linux cpu and single gpu build were chosen arbitrarily to not add too many tests.

[ghstack-poisoned]

PaliC committed:
Update on "[Split Build] Test split build in pull CI workflow"


This PR builds the split build in the pull workflow and runs the appropriate tests against them. A single linux cpu and single gpu build were chosen arbitrarily to not add too many tests.

[ghstack-poisoned]

PaliC committed:
Update on "[Split Build] Test split build in pull CI workflow"


This PR builds the split build in the pull workflow and runs the appropriate tests against them. A single linux cpu and single gpu build were chosen arbitrarily to not add too many tests.

[ghstack-poisoned]

PaliC committed:
Update on "[Split Build] Test split build in pull CI workflow"


This PR builds the split build in the pull workflow and runs the appropriate tests against them. A single linux cpu and single gpu build were chosen arbitrarily to not add too many tests.

[ghstack-poisoned]

PaliC committed:
Update on "[Split Build] Test split build in pull CI workflow"


This PR builds the split build in the pull workflow and runs the appropriate tests against them. A single linux cpu and single gpu build were chosen arbitrarily to not add too many tests.

[ghstack-poisoned]

PaliC committed:
Update

[ghstack-poisoned]

PaliC committed:
Update on "[Split Build] Test split build in pull CI workflow"


This PR builds the split build in the pull workflow and runs the appropriate tests against them. A single linux cpu and single gpu build were chosen arbitrarily to not add too many tests.

[ghstack-poisoned]

PaliC committed:
Update on "[Split Build] Test split build in pull CI workflow"


This PR builds the split build in the pull workflow and runs the appropriate tests against them. A single linux cpu and single gpu build were chosen arbitrarily to not add too many tests.

[ghstack-poisoned]

PaliC committed:
Update on "[Split Build] Test split build in pull CI workflow"


This PR builds the split build in the pull workflow and runs the appropriate tests against them. A single linux cpu and single gpu build were chosen arbitrarily to not add too many tests.

[ghstack-poisoned]

atalman commented on .github/workflows/pull.yml:
```diff
@@ -487,3 +487,31 @@ jobs:
       docker-image: ${{ needs.linux-jammy-py3-clang12-executorch-build.outputs.docker-image }}
       test-matrix: ${{ needs.linux-jammy-py3-clang12-executorch-build.outputs.test-matrix }}
+
+  linux-focal-cuda12_4-py3_10-gcc9-experimental-split-build-build:
```
Name here is ```linux-focal-cuda12_4-py3_10-gcc9-experimental-split-build-build```
However build is ```linux-focal-cuda12.1-py3.10-gcc9-experimental-split-build```

PaliC committed:
Update on "[Split Build] Test split build in pull CI workflow"


This PR builds the split build in the pull workflow and runs the appropriate tests against them. A single linux cpu and single gpu build were chosen arbitrarily to not add too many tests.

[ghstack-poisoned]

PaliC committed:
Update on "[Split Build] Test split build in pull CI workflow"


This PR builds the split build in the pull workflow and runs the appropriate tests against them. A single linux cpu and single gpu build were chosen arbitrarily to not add too many tests.

[ghstack-poisoned]

PaliC committed:
Update on "[Split Build] Test split build in pull CI workflow"


This PR builds the split build in the pull workflow and runs the appropriate tests against them. A single linux cpu and single gpu build were chosen arbitrarily to not add too many tests.

[ghstack-poisoned]

atalman reviewed (approved):
Please fix lint and name, please make sure job is green before merge

PaliC committed:
Update on "[Split Build] Test split build in pull CI workflow"


This PR builds the split build in the pull workflow and runs the appropriate tests against them. A single linux cpu and single gpu build were chosen arbitrarily to not add too many tests.

[ghstack-poisoned]

PaliC committed:
Update on "[Split Build] Test split build in pull CI workflow"


This PR builds the split build in the pull workflow and runs the appropriate tests against them. A single linux cpu and single gpu build were chosen arbitrarily to not add too many tests.

[ghstack-poisoned]

PaliC commented:
@pytorchbot merge

PaliC commented:
@pytorchbort merge -i

PaliC commented:
@pytorchbot merge -i
